File: dsAndAlgo/javascript/bubble_sort.js
function bubble_sort(arr){
  let unsorted_until_index = arr.length - 1
  let sorted = false 

  while (!sorted) {
    sorted = true;
    for (let i = 0; i < unsorted_until_index; i++) {
      if (arr[i] > arr[i + 1]) {
        let temp = arr[i];
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;
        sorted = false
      }
      console.log(unsorted_until_index, arr)
    }
    unsorted_until_index -= 1;
  }
  return arr
}

console.log(bubble_sort([1,5,4,3,2,7,8,5,4,8]))

File: dsAndAlgo/python/bubble_sort.py
def bubble_sort(ls):
    unsorted_until_index = len(ls) - 1
    sorted = False
    
    while not sorted:
        sorted = True
        for i in range(unsorted_until_index):
            if (ls[i] > ls[i+1]):
                ls[i], ls[i+1] = ls[i+1], ls[i]
                sorted = False
        unsorted_until_index -= 1
    return ls

print(bubble_sort([2,3,5,1,6,2,3,7]))